services/event-application-service/src/pubsub_controller/pubsub_subscribers.cc

```diff
@@ -0,0 +1,98 @@
+#include "pubsub_subscribers.h"
+
+#include "glog/logging.h"
+#include "pubsub_controller/pubsub_controller.h"
+#include "queries/domain.h"
+#include "commands/domain.h"
+
+namespace pubsub = ::google::cloud::pubsub;
+
+namespace eas::pubsub_controller {
+
+void CreateFormSub::operator()(google::cloud::pubsub::Message const &msg,
+                               google::cloud::pubsub::AckHandler ack) {
+  // add simple validation.
+  LOG(INFO) << msg;
+  auto cmd = commands::CreateForm{msg.data()};
+  std::move(ack).ack();
+  handler_->Handle(cmd);
+}
+
+void CreateResponseSub::operator()(google::cloud::pubsub::Message const &msg,
+                                   google::cloud::pubsub::AckHandler ack) {
+  // add simple validation.
+  LOG(INFO) << msg;
+  auto cmd = commands::CreateResponse{msg.data()};
+  std::move(ack).ack();
+  handler_->Handle(cmd);
+}
+
+void GetFormSub::operator()(google::cloud::pubsub::Message const &msg,
+                            google::cloud::pubsub::AckHandler ack) {
+  // add simple validation.
+  LOG(INFO) << msg;
+  auto query = queries::FormQuery{msg.data()};
+  std::move(ack).ack();
+  auto result = reader_->Execute(query);
+  controller_->PublishResult(result);
+}
+
+void GetResponseSub::operator()(google::cloud::pubsub::Message const &msg,
+                                google::cloud::pubsub::AckHandler ack) {
+  // add simple validation.
+  LOG(INFO) << msg;
+  auto query = queries::FormQuery{msg.data()};
+  std::move(ack).ack();
+  auto result = reader_->Execute(query);
+  controller_->PublishResult(result);
+}
+
+void GetAllEventResponses::operator()(google::cloud::pubsub::Message const &msg,
+                                      google::cloud::pubsub::AckHandler ack) {
+  // add simple validation.
+  LOG(INFO) << msg;
+  auto query = queries::FormQuery{msg.data()};
+  std::move(ack).ack();
+  auto result = reader_->Execute(query);
+  controller_->PublishResult(result);
+}
+
+void UpdateFormSub::operator()(google::cloud::pubsub::Message const &msg,
+                               google::cloud::pubsub::AckHandler ack) {
+  // add simple validation.
+  LOG(INFO) << msg;
+  auto cmd = commands::UpdateForm{msg.data()};
+  std::move(ack).ack();
+  handler_->Handle(cmd);
+}
+
```

Comment:
I can see only two methods, one with handle, second with PublishResult
The rest seems like duplicates and we need to think how to remove them

Reply:
Each of these `operator()` work with different command object. It calls overridden methods from IQueryHandler/ICommand.

We can think of creating a pretty generic responder. But there's one more issue: each of these structs uniquely identifies a topic to be subscribed.

So, this will be a complete rewrite of the pubsub layer.

Reply:
Well, some kind of 

```cpp
void RegisterSubscriber<std::string_view, Command>();
```

But I'm not sure that this will work.

Reply:
We can create base class for messages with command and query and pass command/query as template parameter or inherit and in constructor assign needed command/query

Reply:
Okay, I will try this, thanks.

Reply:
I'm not sure if it is possible and how much it will cost for us
So it is not strict request, but just nice to have
If you will discover, that it will be hard to implement, then just pass

Reply:
I found an implementation for the idea, thanks for suggestion. It allowed to remove a lot of code.